File: 328-odd-even-linked-list/328-odd-even-linked-list.cpp
class Solution {
public:
    ListNode* oddEvenList(ListNode* head) {
        ListNode* dummy_odd = new ListNode(-1);
        ListNode* dummy_even = new ListNode(-1);
        
        ListNode* odd = dummy_odd;
        ListNode* even = dummy_even;
        
        int n = 0;
        ListNode* curr = head;
        
        while(curr != nullptr){
            n++;
            
            if(n % 2 == 1){
                odd->next = curr;
                odd = odd->next;
            }
            else{
                even->next = curr;
                even = even->next;
            }
            
            curr = curr->next;
        }
        
        even->next = nullptr;
        odd->next = dummy_even->next;
        
        return dummy_odd->next;
    }
};

File: 328-odd-even-linked-list/328-odd-even-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def oddEvenList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        evenSt = []
        oddSt = []
        cur = head
        idx = 1

        while cur:
            if idx%2==0:
                evenSt.append(cur)
            else:
                oddSt.append(cur)
            idx+=1
            cur = cur.next
        point = head = ListNode(0)
        #odd first then even
        while oddSt:
            point.next = oddSt.pop(0)
            point = point.next

        while evenSt:
            point.next = evenSt.pop(0)
            point = point.next

        point.next = None
        return head.next